compiler/modules/front/codeGen.bal

```diff
@@ -363,11 +374,23 @@ function codeGenStmts(CodeGenContext cx, bir:BasicBlock bb, Environment initialE
         else {
             env.assignments.push(...effect.assignments);
         }
+
+        StmtNarrowing[] narrowings = effect.narrowings;
```

Comment:
We need to be clear about the invariant the code is maintaining. Is it that a statement never returns narrowings for variables that were declared within it?

Reply:
Yes, each statement should return narrowing that is to be visible to the next statement. That means compounds statements shouldn't return narrowing internal to them.